salt/utils/templates.py

```diff
@@ -495,9 +495,10 @@ def opt_jinja_env_helper(opts, optname):
                 SLS_ENCODING,
             )
             decoded_context[key] = salt.utils.data.decode(value)
+
+    jinja_env.globals.update(decoded_context)
```

Comment:
**question** why move this here, as opposed to within the `try` block (i.e. after line 500)?

Reply:
Because this cannot cause an exception. Seems to me that the one that causes the exception is `template = jinja_env.from_string(tmplstr)`, for example when the template contains a bug. That is why before was part of the `try` block.

Reply:
Do you see this as a blocker?  I can see it giving an extra level of protection here in the event of an exception happening at this time, but both of these are dictionaries which are created above, so we can be reasonably sure they always will be.

Reply:
> I can see it giving an extra level of protection here in the event of an exception happening at this time

`dict.update()` cannot produce an exception at this point ... We should not be confused that this update was before in the `try` just because was using the `template` variable (that can produce an exception during the creation, if the template has a bug)